### PR TITLE
[IMP] account: Faster computation of cash basis

### PR DESCRIPTION
**Pull request by yajo**
Translates slow fields to SQL:

- matched_percentage
- debit_cash_basis
- credit_cash_basis
- balance_cash_basis

@Tecnativa TT18838

### CODE REVIEW

**Comment by yajo** [comment]
@pedrobaeza please review.

**Comment by pedrobaeza** [comment]
Please check Travis. You have to convert second argument of ROUND to integer

**Comment by yajo** [comment]
Weird, `CEIL()` was supposed to return integer... Well, casted anyways.

File: addons/account/migrations/9.0.1.1/post-migration.py [hunk 1]
```diff
@@ -974,6 +974,80 @@ def fill_move_line_invoice(cr):
...
+    openupgrade.logged_query(
+        env.cr,
+        """
+            UPDATE account_move
+            SET matched_percentage = sub.matched_percentage
+            FROM (
+                SELECT
+                    aml.move_id,
+                    CASE
+                        WHEN 0.0 == ROUND(
```
Comment (pedrobaeza) [inline addons/account/migrations/9.0.1.1/post-migration.py]: This is not Python, hehe

```suggestion
                        WHEN 0.0 = ROUND(
```

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**